api.go

```diff
@@ -194,39 +206,76 @@ func (c *Client) Durations() ([]int, error) {
 		err = decodeBody(resp, &durationErr)
 		if err != nil {
 			if reqID := GetRequestID(resp); reqID != "" {
-				return nil, fmt.Errorf(ParseErrorReqId, reqID, err)
+				return nil, &AlksError{
+					StatusCode: resp.StatusCode,
+					RequestId:  "",
+					Err:        fmt.Errorf(ParseErrorReqId, reqID, err),
+				}
 			}
 
-			return nil, fmt.Errorf(ParseError, err)
+			return nil, &AlksError{
+				StatusCode: resp.StatusCode,
+				RequestId:  "",
+				Err:        fmt.Errorf(ParseError, err),
+			}
 		}
 
 		if durationErr.Errors != nil {
 			if reqID := GetRequestID(resp); reqID != "" {
-				return nil, fmt.Errorf(ErrorStringFull, reqID, resp.StatusCode, durationErr.Errors)
+				return nil, &AlksError{
+					StatusCode: resp.StatusCode,
+					RequestId:  reqID,
+					Err:        fmt.Errorf(ErrorStringFull, reqID, resp.StatusCode, durationErr.Errors),
+				}
 			}
 
-			return nil, fmt.Errorf(ErrorStringNoReqId, resp.StatusCode, durationErr.Errors)
+			return nil, &AlksError{
+				StatusCode: resp.StatusCode,
+				RequestId:  "",
+				Err:        fmt.Errorf(ErrorStringNoReqId, resp.StatusCode, durationErr.Errors),
+			}
 		}
 
 		if reqID := GetRequestID(resp); reqID != "" {
-			return nil, fmt.Errorf(ErrorStringOnlyCodeAndReqId, reqID, resp.StatusCode)
+			return nil, &AlksError{
+				StatusCode: resp.StatusCode,
+				RequestId:  reqID,
+				Err:        fmt.Errorf(ErrorStringOnlyCodeAndReqId, reqID, resp.StatusCode),
+			}
 		}
 
-		return nil, fmt.Errorf(ErrorStringOnlyCode, resp.StatusCode)
+		return nil, &AlksError{
+			StatusCode: resp.StatusCode,
+			RequestId:  "",
+			Err:        fmt.Errorf(ErrorStringOnlyCode, resp.StatusCode),
+		}
 	}
 
 	lrr := new(LoginRoleResponse)
 	err = decodeBody(resp, &lrr)
 	if err != nil {
 		if reqID := GetRequestID(resp); reqID != "" {
-			return nil, fmt.Errorf("Error parsing LoginRole response: [%s] %s", reqID, err)
+			return nil, &AlksError{
+				StatusCode: 0,
+				RequestId:  reqID,
+				Err:        fmt.Errorf("Error parsing LoginRole response: [%s] %s", reqID, err),
+			}
+
 		}
 
-		return nil, fmt.Errorf("Error parsing LoginRole response: %s", err)
+		return nil, &AlksError{
+			StatusCode: 0,
+			RequestId:  "",
+			Err:        fmt.Errorf("Error parsing LoginRole response: %s", err),
+		}
 	}
 
 	if lrr.RequestFailed() {
-		return nil, fmt.Errorf("Error fetching role information: [%s] %s", lrr.BaseResponse.RequestID, strings.Join(lrr.GetErrors(), ", "))
+		return nil, &AlksError{
+			StatusCode: 0,
```

Comment:
Why are we not returning a status code in this block of all these different files?